crates/ra_hir/src/ty/primitive.rs

```diff
@@ -18,10 +18,7 @@ impl From<IntTy> for Uncertain<IntTy> {
 
 impl fmt::Display for Uncertain<IntTy> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            Uncertain::Unknown => write!(f, "{{integer}}"),
-            Uncertain::Known(ty) => write!(f, "{}", ty),
-        }
+        write!(f, "{}", self.ty_to_string())
```

Comment:
Using `&'static str` in the trait isn't that nice, what was wrong with the old match approach? And that could be combined with a `write_display` that takes a `&impl fmt::Display`, since these types already implement `fmt::Display`.

Reply:
I just maintained existing function signatures for a minimal diff.